nixos/modules/services/backup/bupstash.nix

```
@@ -0,0 +1,556 @@
+              };
+
+              keys = mkOption {
+                type = listOf str;
+                default = [ ];
+                description = "List of ssh keys that can be used to access this repository.";
+              };
+
+              allowed = mkOption {
+                type = listOf (
+                  enum [
+                    "init"
+                    "put"
+                    "list"
+                    "get"
+                    "remove"
+                    "gc"
+                    "sync"
+                  ]
+                );
+                default = [ ];
+                description = ''
+                  List of allowed bupstash actions through the ssh connection.
+                  If none are specified then all actions are authorized.
+                '';
+              };
+            };
+          }
+        );
+        default = { };
+        description = "List of ssh keys with access to the repository.";
+      };
+
+      home = mkOption {
+        type = passwdEntry path;
+        default = "/var/lib/${defaultRepositoryUser}";
+        description = "The path where the repositories will be stored.";
+      };
+
+      restrictCommands = mkOption {
+        type = bool;
+        default = true;
+        description = "If set to true, restricts the commands that can be run through ssh by the user to `bupstash ...`.";
+      };
+    };
+
+    jobs = mkOption {
+      type = attrsOf (
+        submodule (
+          { name, ... }:
+          {
+            options = {
+              paths = mkOption {
+                type = listOf path;
+                default = [ ];
+                description = "List of paths to backup.";
+              };
+
+              excludes = mkOption {
+                type = listOf str;
+                default = [ ];
+                description = "List of patterns to exlude from the backup.";
+              };
+
+              command = mkOption {
+                type = nullOr str;
+                default = null;
+                description = "Command used for getting the backup data.";
+              };
+
+              key = mkOption {
+                type = nullOr path;
+                default = null;
+                description = "Path to the key that will be used to encrypt the data.";
+              };
+
+              keyCommand = mkOption {
+                type = nullOr str;
+                default = null;
+                description = ''
+                  A command to run that must print the key data, can be used instead of `job.key`
+                  to fetch the key from arbitrary locations such as the network or other secret storage.
+                '';
+              };
+
+              user = mkOption {
+                type = str;
+                default = defaultClientUser;
+                description = ''
+                  User for the backup job. The default user '${defaultClientUser}' is not created, so it is only cosmetic.
```

```suggestion
                  User that executes the backup job. Will run as a [dynamic user](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=) by default.
                  If the backup needs to run under a static UID, declaring a user account with the same name is                   sufficient.
```

I don't think we need to put all the reasoning in there.